src/helpers/datetime.js

```diff
@@ -21,3 +21,30 @@ export function formatDate(formatString, date) {
 
   return moment(date || new Date()).format(formatString);
 }
+
+/**
+ * A formatLocaleDate helper to format date with locale using moment js.
+ *
+ * @example
+ *      {{formatDate 'MM/DD/YYYY' 'es' date}}
+ *
+ * @param {string} formatString Based on moment.js.
+ * @param {string} localeString Language or language-country locale string available in https://github.com/moment/moment/tree/develop/locale .
+ * @param {Date} date
+ * @returns {string}
+ */
+export function formatLocaleDate(formatString, localeString, date) {
+  let moment = global.moment;
+
+  if (!moment) {
+    moment = require('moment');
+  }
+
+  formatString = isString(formatString) ? formatString : '';
+
+  if (isString(localeString)) {
+    moment.locale(localeString);
```

Comment:
Does this operation only change the locale for the next execution or will it set the locale globally?

Reply:
Yes, looks like it does it in the global context, thus making this function not pure. 

Can you try something like this to apply the locale just for this instance locally? Check this out - https://momentjs.com/docs/#/i18n/instance-locale/

Reply:
Would we want to create a new locale every time this helper function is hit or store a moment instance per-locale somewhere (essentially cache them as-needed)? Maybe do something like store them as `global.localmoment.fr`, `global.localmoment.es`, etc and pull them as-needed.

Reply:
@rhanton If you want to do it globally tthen you could still do it separately before initializing the helpers in your own JS script like `moment.locale(localeString);`. The helper function shouldn't be modifed for that. 

For the helper's use case, it only make sense to have local locale when parameter is passed, since every helper call should exhibit an isolated and predictable behavior.

Reply:
What do you think about a global `setDateLocale` helper as well that does modify the configuration of the global.moment? This would provide users two options: passing locale to `formatDate` each time, or the `setDateLocale` helper that affects all future `formatDate` calls? For us, the latter would actually be helpful, as we have a situation where it is likely that someone will change a handlebars template in the future, not realizing that because it is for messages in Spanish, they must designate 'es' in `formatDate` - whereas having a global locale designation would prevent this issue. Thoughts?

Reply:
Okay, I reworked the code to work this way, have a look and tell me if that's a bad idea to all users to change the global moment locale (or any other comments etc).

Reply:
@rhanton I see your point here. The changes you did for the `formatDate` function looks good. 
But I don't see a need for `setDateLocale` function here - that is simply not how templating engines like handlerbars should behave. 

That is - these helpers should only care about displaying things in the UI but shouldn't change anything for your app. For your use case you should set the global locale directly from your JS and let the helper display date as per the format and locale set - which should just work fine for both global and locale settings. See my other comment. 

This makes the behavior predictable as well.

Reply:
Sounds good.